### ISSUE TITLE
windows_web_app / linux_web_app - Error with win32_status even when argument is not used.

### ISSUE THREAD

**Issue by calebak404** [hashicorp/terraform-provider-azurerm#22964]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.3.7

### AzureRM Provider Version

3.63

### Affected Resource(s)/Data Source(s)

azurerm_windows_web_app, azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "example" {
  name                = "example"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_service_plan.example.location
  service_plan_id     = azurerm_service_plan.example.id

  site_config {
    auto_heal_setting {
      action {
        action_type = "Recycle"
        minimum_process_execution_time = "05:00:00"
      }
      trigger {
        count = 4
        interval = "00:10:00"
        path = "true"
        status_code_range = "500-599"
      }
  }
}
```

### Debug Output/Panic Output

```shell
Error: a number is required
```

### Actual Behaviour

#22003 included in 3.63 added a breaking change modifying the type of `win32_status` from string to int.

This change affects any environment using auto-heal, even if the `win32_status` value is not configured.

Setting the `win32_status` value to a number in the configuration does not resolve the issue, and it appears to require manual state editing to resolve.

### Steps to Reproduce

1. With azurerm 3.62.1, deploy a web app with an auto_heal rule that does not use win32_status
2. Update to azurerm 3.63 or newer

**Comment by xiaxyi** [comment]
Thanks @calebak404 for raising this issue, `win32_status` is defined as `Optional`, so it shouldn't be required during the resource creation, below is the code definition:
```
"win32_status": {
Type:     pluginsdk.TypeInt,
Optional: true,
}
```

Also, I can run below config without any issues:
```
resource "azurerm_linux_web_app" "example" {
  name                = "xiaxintest-lwa"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_service_plan.test.location
  service_plan_id     = azurerm_service_plan.test.id

  site_config {
    auto_heal_enabled = true
    auto_heal_setting {
      action {
        action_type                    = "Recycle"
        minimum_process_execution_time = "05:00:00"
      }
      trigger {
        status_code {

          count             = 4
          interval          = "00:10:00"
          path              = "true"
          status_code_range = "500-599"
        }
      }
    }
  }
}
```
You may want to add the status_code {} block to the properties, otherwise, you will get the error
```
count             = 4
          interval          = "00:10:00"
          path              = "true"
          status_code_range = "500-599"
```

**Comment by calebak404** [comment]
We use the status_code block and do not set win32_status

This is an issue for all of our environments that had this configuration deployed with provider version < 3.63 when they try to use 3.63

**Comment by calebak404** [comment]
The issue is that even though it is optional, the value gets set in state.
3.62.1 and older set it as ""
3.63.0 and newer set it as 0

Because the provider is expecting it to be a number in 3.63.0 and newer, it errors out when it sees the empty string from resources deployed prior to 3.63.0

**Comment by rcskosir** [comment]
Thank you for opening this issue. Linking to a related PR: #23075

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. I am going to mark this issue as closed with #23075. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
